### PR TITLE
Expiry date

### LINKED ISSUE #12: Root policy: expiration
The expires section is validated as being correct (not in the past)

### PR DESCRIPTION
**Pull request by jyotsna-penumaka**
Resolves #12 

#### Ticket Link

Fixes

#### Release Note

```release-note

```

### CODE REVIEW

**Review by lkatalin — APPROVED** [review]
LGTM!

File: src/policy.rs [hunk 1]
```diff
@@ -103,4 +103,38 @@ fn parse_script_success() {
...
+
+    let mut file = File::open(&*dir.to_string_lossy()).unwrap();
+    let mut contents = String::new();
+    file.read_to_string(&mut contents).unwrap();
+
+    let policy: Policy = serde_json::from_str(&contents).unwrap();
+    let expiry = policy.signed.expires;
+    let current = Utc::now();
+
+    let duration = expiry.signed_duration_since(current);
```
Comment (asraa) [inline src/policy.rs]: should we move this into a ValidateExpires function?

**Review by asraa — COMMENTED** [review]
Beautiful!!!!

File: src/policy.rs (hunk 1 above)
Comment (jyotsna-penumaka) [reply src/policy.rs]: That's a great idea.
I will make that change now.

File: src/policy.rs [hunk 2]
```diff
@@ -92,6 +92,13 @@ pub struct SigstoreOidcKey {
     pub issuer: String,
 }
 
+fn validate_expires(contents: String) -> chrono::Duration {
```
Comment (asraa) [inline src/policy.rs]: sorry one more nit: maybe the param should be a policy? so we can

```
 let policy: Policy = serde_json::from_str(&contents).unwrap();
 let duration = validate_expires(policy);
 assert_eq!(duration.to_std().is_err(), false);)
 // run other validations on policy (and not re-parse)
```

**Review by asraa — CHANGES_REQUESTED** [review]

File: src/policy.rs (hunk 2 above)
Comment (jyotsna-penumaka) [reply src/policy.rs]: sounds good!

**Review by asraa — APPROVED** [review]
Thanks!!

**Comment by lukehinds** [comment]
sorry @jyotsna-penumaka just need a rebase!

**Review by lukehinds — APPROVED** [review]

**Comment by imjasonh** [comment]
I get an error with `cargo build` and `cargo run` with this change, since `validate_expires` isn't used except in tests (`cargo test` works). Is there more work planned here to wire this up to the rest of the `sget` CLI? Anything I can help with?

**Pull request merged**